### DISCUSSION TITLE
Some styles don't work

### DISCUSSION THREAD

**Discussion by angelovdev** [tailwindlabs/tailwindcss#5963] [Help] [answered]
**What version of Tailwind CSS are you using?**

2.2.15 and tried 2.2.19

**What build tool (or framework if it abstracts the build tool) are you using?**

postcss-cli 8.3.6, Next.js 11.1.2, webpack 5*

**What version of Node.js are you using?**

v16.11.1

**What browser are you using?**

Chromium/Firefox

**What operating system are you using?**

Linux

**Reproduction URL**

N/A as I'd have to put the whole project public to reproduce it.

<details>
<summary>tailwind.config.js</summary>

```js
module.exports = {
  mode: 'jit',
  purge: ['./pages/**/*.{js,ts,jsx,tsx}', './components/**/*.{js,ts,jsx,tsx}'],
  darkMode: false, // or 'media' or 'class'
  theme: {
    extend: {},
    fontFamily: {
      sans: ['Muli', 'sans-serif'],
      serif: ['serif'],
    },
    colors: {
      transparent: 'transparent',
      current: 'currentColor',
      black: {
        DEFAULT: '#000000',
      },
      gray: {
        lighter: '#FAFAFA',
        DEFAULT: '#E8E7EA',
        darker: '#C8C8C8',
      },
      blue: {
        lighter: '#AFD8F8',
        light: '#56B4FC',
        DEFAULT: '#0052FF',
      },
      green: {
        DEFAULT: '#7BA741',
      },
      red: {
        lighter: '#F5C5BD',
        light: '#ED9B8C',
        DEFAULT: '#E1543B',
      },
      orange: {
        DEFAULT: '#F1AB41',
      },
      purple: {
        DEFAULT: '#6D06CF',
      },
      white: {
        DEFAULT: '#FFFFFF',
      },
    },
  },
  variants: {
    extend: {},
  },
  plugins: [],
};
```

</details>

<details>
<summary>postcss.config.js</summary>

```js
module.exports = {
  plugins: {
    'tailwindcss/nesting': {},
    tailwindcss: {},
    autoprefixer: {},
  },
};
```

</details>

<details>
<summary>next.config.js</summary>

```js
/** @type {import('next').NextConfig} */
const { i18n } = require('./next-i18next.config');

const { withSentryConfig } = require('@sentry/nextjs');

const moduleExports = {
  reactStrictMode: true,
  i18n,
  basePath: '',
  async rewrites() {
    return {
      fallback: [
        {
          source: '/:path*',
          destination: `[redacted]:path*`,
          locale: false,
        },
      ],
    };
  },
  async redirects() {
    return [
      {
        source: '/[redacted]',
        basePath: false,
        destination: '[redacted]',
        permanent: false,
      },
      {
        source: '/[redacted]',
        destination: '/[redacted]',
        permanent: true,
      },
    ];
  },
  //https://github.com/tailwindlabs/tailwindcss/issues/5107
  webpack(config) {
    const rules = config.module.rules
      .find((rule) => typeof rule.oneOf === 'object')
      .oneOf.filter((rule) => Array.isArray(rule.use));

    rules.forEach((rule) => {
      rule.use.forEach((moduleLoader) => {
        if (moduleLoader.loader.includes('resolve-url-loader'))
          moduleLoader.options.sourceMap = false;
      });
    });

    return config;
  },
};

const SentryWebpackPluginOptions = {
  silent: true, // Suppresses all logs
};

module.exports = withSentryConfig(moduleExports, SentryWebpackPluginOptions);
```

</details>

**Describe your issue**

Some of the styles applied through classes are not working.

Those 2 elements are the issue.

![image](https://user-images.githubusercontent.com/71898342/139817050-d7eb7023-d01e-41f2-b935-233affb31dc1.png)

Styles of the `<ul>` in the browser (missing margin)
![image](https://user-images.githubusercontent.com/71898342/139817231-e0ce9366-95be-42e0-9ddc-6e84b744335a.png)

Styles of the `<a>` in the browser (missing rounded-xl and bg-blue-lighter)
![image](https://user-images.githubusercontent.com/71898342/139817460-c91ee527-6546-483d-b47d-11fd32588c09.png)

Changing the padding to a different valid value:
![image](https://user-images.githubusercontent.com/71898342/139817701-1acd1010-8bcb-497f-9f3d-c82fc1b8b810.png)

Style disappears completely:
![image](https://user-images.githubusercontent.com/71898342/139817769-ef982b97-4692-4a2e-9ba6-d7d433312262.png)

It's a project with about 10 pages and 30 components. This issue isn't happening on any other page or component.

**Comment by adamwathan** [comment, marked as answer]
My guess is you are missing a path in your `purge` config but without a reproduction it’s impossible to know.

You mentioned the project has components but your `purge` config only includes the `pages` folder, do you have a `components` folder too you need to add?

**Reply by angelovdev** [reply]
My tailwind.config.js file includes the components folder but thanks for pointing me in the direction. 

This code was part of a layout .tsx file stored in a `./layouts` folder which I was missing in my purge array so only the styles that I have used elsewhere were working.

Thanks again!
